# Genetic mapping of fitness determinants across the malaria parasite Plasmodium falciparum life cycle

Xue Li, Sudhir Kumar, Marina McDew-White, Meseret Haile, Ian H. Cheeseman, Scott Emrich, Katie Button-Simons, François Nosten, Stefan H.I. Kappe, Michael T. Ferdig, Tim J.C. Anderson and Ashley M. Vaughan

Abstract
Determining the genetic basis of fitness is central to understanding evolution and transmission of microbial pathogens. In human malaria parasites (Plasmodium falciparum), most experimental work on fitness has focused on asexual blood stage parasites, because this stage can be easily cultured, although the transmission of malaria requires both female Anopheles mosquitoes and vertebrate hosts. We explore a powerful approach to identify the genetic determinants of parasite fitness across both invertebrate and vertebrate life-cycle stages of the human malaria parasite P. falciparum. This combines experimental genetic crosses using humanized mice, with selective whole genome amplification and pooled sequencing to determine genome-wide allele frequencies and identify genomic regions under selection across multiple lifecycle stages. We applied this approach to genetic crosses between artemisinin resistant (ART-R, kelch13-C580Y) and ART-sensitive (ART-S, kelch13-WT) parasites, recently isolated from Southeast Asian patients. Two striking results emerge: we observed (i) a strong genome-wide skew (>80%) towards alleles from the ART-R parent in the mosquito stage, that dropped to ~50% in the blood stage as selfed ART-R parasites were selected against; and (ii) repeatable allele specific skews in blood stage parasites with particularly strong selection (selection coefficient (s) ≤ 0.18/asexual cycle) against alleles from the ART-R parent at loci on chromosome 12 containing MRP2 and chromosome 14 containing ARPS10. This approach robustly identifies selected loci and has strong potential for identifying parasite genes that interact with the mosquito vector or compensatory loci involved in drug resistance.

